### PR TITLE
Enable MacOs vulkan test on CI

### PR DESCRIPTION
**Pull request by PhaneeshB**
xfail tests not working
enable MacStudio Vulkan tests

### CODE REVIEW

**Review by monorimet — APPROVED** [review]
LGTM! Nice.

File: tank/test_models.py [hunk 1]
```diff
@@ -365,6 +365,33 @@ def test_module(self, dynamic, device, config):
...
             pytest.xfail(reason="Transformers API mismatch")
+        if config["model_name"] == "alexnet" and device in ["metal", "vulkan"]:
+            pytest.xfail(reason="Assertion Error: Zeros Output")
+        if (
+            config["model_name"] == "camembert-base"
+            and dynamic == False
+            and device in ["metal", "vulkan"]
+        ):
+            pytest.xfail(
+                reason="chlo.broadcast_compare failed to satify constraint"
```
Comment (dan-garvey) [inline tank/test_models.py:376]: Please link to upstream issue if available

File: tank/test_models.py [hunk 2]
```diff
@@ -365,6 +365,33 @@ def test_module(self, dynamic, device, config):
...
+            )
+        if (
+            config["model_name"] == "roberta-base"
+            and dynamic == False
+            and device in ["metal", "vulkan"]
+        ):
+            pytest.xfail(
+                reason="chlo.broadcast_compare failed to satify constraint"
```
Comment (dan-garvey) [inline tank/test_models.py:384]: same as above

File: tank/test_models.py [hunk 3]
```diff
@@ -365,6 +365,33 @@ def test_module(self, dynamic, device, config):
...
+            )
+        if config["model_name"] in [
+            "microsoft/MiniLM-L12-H384-uncased",
+            "wide_resnet50_2",
+            "resnet50",
+            "resnet18",
+            "resnet101",
+            "microsoft/resnet-50",
+        ] and device in ["metal", "vulkan"]:
+            pytest.xfail(reason="Vulkan Numerical Error (mostly conv)")
```
Comment (dan-garvey) [inline tank/test_models.py:394]: same as above

**Pull request merged**